rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -257,7 +273,9 @@ Node::get_parameter_or_set(
   ParameterT & value,
   const ParameterT & alternative_value)
 {
-  bool got_parameter = get_parameter(name, value);
+  std::string extended_name = get_extended_name(name);
+
+  bool got_parameter = get_parameter(extended_name, value);
```

Comment:
I'm not sure about this for parameter names... I think they use the `.` separator at the moment if I'm not mistaken (https://github.com/ros2/rclpy/blob/5b95169e6f4bc05e3052e22ccbe036df69898f8a/rclpy/rclpy/parameter.py#L20), though I think that's been a point of discussion in the past.

@tfoote @dirk-thomas can I get a consult on this from one or both of you guys? Specifically "should a concept like `NodeHandle` from ROS 1 affect the name when getting or setting a parameter"?